src/routes/auth.ts

```diff
@@ -0,0 +1,60 @@
+import { Router, Request, Response, NextFunction } from "express";
+import dotenv from "dotenv";
+import passport from "passport";
+import { OAuthStrategy, VerifyFunction } from "passport-google-oauth";
+import { IProfile } from "../models/User.d";
+import { User } from "../models/User";
+
+dotenv.config();
+
+const { consumerKey, consumerSecret, callbackURL } = process.env as {
+  [key: string]: string;
+};
+
+const router = Router();
+
+passport.serializeUser((user, cb) => {
+  cb(null, user);
+});
+
+passport.deserializeUser((user, cb) => {
+  cb(null, user);
+});
+
+passport.use(
+  new OAuthStrategy(
+    {
+      consumerKey: consumerKey,
```

Comment:
If we have the same name for key and value in object, we can shorthand it

```suggestion
      consumerKey,
```